### ISSUE TITLE
Analyzer message NUnit2005 is indeed extremely annoying

### ISSUE THREAD

**Issue by makemefeelgr8** [nunit/nunit#4661]
You closed this ticket https://github.com/nunit/nunit/issues/4401 without actually addressing it. Let me bring it to your attention once again. Who even thought that making every test throw a warning was a good idea?? And a warning is like "consider using this new great feature". Like, does your iphone throw a warning when they release a new one? Like, "Consider using IPhone 16 now!"? 

Does the main function throw a warning, now that there's a top level statement pattern?
Does a controller throa a warning, now that there's a minimal api approach?

`Assert.AreEqual` was industry standard for years, and it still continues to be. It is the default.

So, I'm asking you loud and clear:

**How do I make every assert line not produce a warning?**

Things I'm not going to do:
1. Use the new assert model. I considered it. I like the old one better. I don't feel like spending months to rewrite 200,000 existing tests in my project.
2. Modify 200000 tests to use a "classic assert" statement for obvious reasons.
3. Do some weird trickery like having to figure out what is a ruleset. And then figure out how to create my own thing. And then to add it to every one of a 900 projects. If there was a manual, like- ok, download this file, place it in here or something... But this is just ridiculous!
4. I don't want to write a custom `.editorconfig` files and commit them. This sucks. There's a reason those files don't belong in a repo. Everybody has a different `.editorconfig` locally!
5. Adding some weird "pragma directive" to every line of code that contains `assert` in every file. No comments here. It's obvious. This is just stupid.
6. Adding `attribute [SuppressMessage]`. I'm not gonna do this, as I don't know where to add it. And this [joke of a doc](https://github.com/nunit/nunit.analyzers/blob/master/documentation/NUnit2005.md) says nothing about it.

Things I am considering to do:

1. Stick to NUnit v3 forever
2. Switch to XUnit / MStest once it gets discontinued

Fix this shit. Like, seriously. Fix this. Please, guys. This is stupid. 
There has to be a better way of solving this, than just saying "shut up, screw you, Imma close this thing as a duplicate".

**Comment by manfred-brands** [comment]
If you want to use the `Assert.AreEqual` that is up to you.

Note that NUnit2005 would also fire for NUnit v3 and has done so for years.

### Disabling NUnit2005 :
As mentioned in #4401, if you don't like what the analyzer suggests, remove it.

If you only don't like this one rule, disable it. 
There is microsoft [documentation](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/configuration-files) for these, but in summary, there are 2 options:
1. Add a single .editorconfig in the root directory of all your projects containing:
```
root = true

[*.cs]

# NUnit2005: Consider using Assert.That(actual, Is.EqualTo(expected)) instead of ClassicAssert.AreEqual(expected, actual)
dotnet_diagnostic.NUnit2005.severity = none
```
2. Create a "DisableNUnit2005.globalconfig" file containing:
```
is_global = true 

# NUnit2005: Consider using Assert.That(actual, Is.EqualTo(expected)) instead of ClassicAssert.AreEqual(expected, actual)
dotnet_diagnostic.NUnit2005.severity = none
```
and reference it in your `Directory.Build.props` file for your solution like in:
```
<ItemGroup>
    <GlobalAnalyzerConfigFiles Include="$(MSBuildThisFileDirectory)/DisableNUnit2005.globalconfig"" />
</ItemGroup>
```

> I don't want to write a custom .editorconfig files and commit them. This sucks. There's a reason those files don't belong in a repo. Everybody has a different .editorconfig locally!

I disagree here, they DO belong in a repo. It ensures consistent rules and style for everybody using the repo.
You want to disable the NUnit2005 for everybody using your project not just you isn't it?
For my work, we use one .editorconfig in a repo containing 500+ projects shared with 60+ developers.
If each of them used a different coding style and other rules it would be a mess.

### If you are talking about it not compiling that is another issue.

However, to make them compile with NUnit 4 you needs to add something to your solution only once in the root directory:
See the [migration guide](https://docs.nunit.org/articles/nunit/release-notes/Nunit4.0-MigrationGuide.html#use-global-using-aliases)

**Comment by OsirisTerje** [comment]
> Fix this shit. Like, seriously. Fix this. Please, guys. This is stupid.
There has to be a better way of solving this, than just saying "shut up, screw you, Imma close this thing as a duplicate".

and

> And this [joke of a doc](https://github.com/nunit/nunit.analyzers/blob/master/documentation/NUnit2005.md) says

You may be as angry as you like, but taking it out on us this way, using language, is no acceptable behavior.  Please see https://dotnetfoundation.org/about/policies/code-of-conduct .  See the note at the end of this issue, saying "Remember, contributions to this repository should follow its [contributing guidelines](https://github.com/nunit/nunit/blob/f4f266b4c0ddebc708c2189fa17b24a0fdc14730/CONTRIBUTING.md) and [code of conduct](https://github.com/nunit/nunit/blob/f4f266b4c0ddebc708c2189fa17b24a0fdc14730/CODE_OF_CONDUCT.md).". This also applies to issues you raise.

We are doing this work on a voluntary non-paid basis, and we are not into this to be called "stupid", "shit" and "joke". 

To address some of your items:

We have written [migration documentation](https://docs.nunit.org/articles/nunit/release-notes/Nunit4.0-MigrationGuide.html) to the best of our abilities, which also includes ways to avoid what you write above about how to continue using the classic asserts using only `Assert`, see [this section](https://docs.nunit.org/articles/nunit/release-notes/Nunit4.0-MigrationGuide.html#use-global-using-aliases) for a couple of suggestions.

This fixes your point 1 and 2 above.

Point 3, not sure why you would need a ruleset.

Point 4: It actually is common to include an .editorconfig committed to the repo.  That would constitute how your team would like to handle the code in the projects, so it is not a bad idea, imho.  Having only locals of this file would easily cause files to change between commits from different developers, and could be seen as annoying.  

Point 5. Agree, I would not recommend using pragmas on anything but special cases. 

Point 6.  Execpt for the "joke" comment, I agree that this is equal to point 5, only for special cases. 

So to solve you issue with the analyzer, and given you still dont want to add suppressions to .editorconfig, you can add the suppressions a [global suppressions file](https://learn.microsoft.com/en-us/visualstudio/code-quality/in-source-suppression-overview?view=vs-2022&tabs=csharp#suppress-violations-using-a-global-suppression-file) named `GlobalSuppressions.cs`.

Alternatively, skip installing the analyzer package altogether.

Staying on the version 3.X **IS** a valid option.  As said in other responses, we **will** maintain that series and backport changes as far as we are able to - in our free time.  

PS: Noted @manfred-brands also answered this at the same time.

**Comment by makemefeelgr8** [comment]
> if you don't like what the analyzer suggests, remove it.

Thanks, this is just what I need.

> Note that NUnit2005 would also fire for NUnit v3 and has done so for years.

Well, this beast of a solution is using some pretty outdated libraries. Has been using till now.

> You may be as angry as you like

Nah nah sorry man, I'm no angry. Adding some emotion to it is just a way to get someone to react. I mean, take a look at [this poor soul](https://github.com/nunit/nunit/issues/4143) - the wrote a polite ticket in accordance to all codes and conducts and got ignored for 2 years.

Thanks for the migration doc. Somehow google searches didn't show that one, but instead pointed to this [weird doc](https://github.com/nunit/nunit.analyzers/blob/master/documentation/NUnit2005.md). I wonder if one could just add the-migration-link there.

**Comment by makemefeelgr8** [comment]
To anybody from the future who comes in here and sees this, the easiest option to make it disappear is to just stick the following into your `.csproj`:
```xml
<PropertyGroup><NoWarn>NUnit2005</NoWarn></PropertyGroup>
```
